unix-opts.lisp

```diff
@@ -462,32 +462,57 @@ to `nil')"
                (process-option item))
               (t (push item free-args)))))))
 
-(defun add-text-padding (str &key padding newline)
-  "Add padding to text STR. Every line except for the first one, will be
-prefixed with PADDING spaces. If NEWLINE is non-NIL, newline character will
-be prepended to the text making it start on the next line with padding
+(defun add-text-padding (str &key padding newline max-width)
+  "Add padding to text STR. Every line except for the first one, will
+be prefixed with PADDING spaces, and lines will be wrapped before
+MAX-WIDTH characters. If NEWLINE is non-NIL, newline character will be
+prepended to the text making it start on the next line with padding
 applied to every single line."
-  (let ((pad            (make-string padding      :initial-element #\Space))
-        (pad-next-lines (make-string (max 0 (1- padding)) :initial-element #\Space)))
-    (with-output-to-string (s)
-      (when newline
-        (format s "~%~a" pad))
-      (map nil
-           (lambda (x)
-             (write-char x s)
-             (when (char= x #\Newline)
-               (write pad-next-lines :stream s :escape nil)))
-           str))))
-
-(defun print-opts (defined-options &optional (stream *standard-output*) (argument-block-width 25))
-  "Print info about defined options to STREAM. Every option get its own line
-with description. A newline is printed after the options if this part of the
-text is wider than ARGUMENT-BLOCK-WIDTH."
+  (with-output-to-string (s)
+    (when newline
+      (format s "~%~vt" padding))
+    (loop for c across str
```

Comment:
This way of looping over each character and calling `line-break-needed-p` for each character is quite inefficient as we are looking forward for the next whitespace char for every character. I've added a more efficient suggestion in the review.